cdap-data-fabric/src/main/java/co/cask/cdap/store/NamespaceMetadataStore.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.store;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.data2.dataset2.lib.table.MDSKey;
+import co.cask.cdap.data2.dataset2.lib.table.MetadataStoreDataset;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.NamespaceMeta;
+import co.cask.cdap.proto.id.NamespaceId;
+import com.google.gson.Gson;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Store for namespace metadata
+ */
+public class NamespaceMetadataStore extends MetadataStoreDataset {
+
+  private static final Gson GSON = new Gson();
+  private static final String TYPE_NAMESPACE = "namespace";
+
+  public NamespaceMetadataStore(Table table) {
+    super(table);
+  }
+
+  @Override
+  protected <T> byte[] serialize(T value) {
```

Comment:
Don't remember why I overrode these methods, but looks like they're the same as the parent methods. Removed.